aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -35,18 +91,155 @@ struct _cpu_scatter_gather_dim_loop {
       );
     }
   }
+
+  template <typename scalar_t, typename func_t>
+  void operator()(
+    scalar_t* self_data, int64_t self_dim_stride,
+    int64_t* index_data, int64_t index_dim_stride,
+    Scalar value,
+    int64_t dim, int64_t index_dim_size,
+    int64_t index_upper_bound,
+    func_t f
+  ) {
+
+    for (int64_t i = 0; i < index_dim_size; ++i) {
+      int64_t idx_dim = index_data[i * index_dim_stride];
+      // we are not putting idx_dim in the error message because it disables
+      // loop optimization in clang-7
+      TORCH_CHECK(idx_dim >= 0 && idx_dim < index_upper_bound,
+        "index ", index_data[i * index_dim_stride],
+        " is out of bounds for dimension ", dim,
+        " with size ", index_upper_bound
+      );
+      auto temp = value.to<scalar_t>();
```

Comment:
nit: Not sure that `temp` is a good name here...

Reply:
Well its just holding the value until it is passed into the function...

Reply:
Yes, but the value that this temp holds lives until the end of the execution, and the name temp suggests that it lives for a very short time...

Reply:
End of the execution? Its only until the end of the next statement (which is a call to the kernel which is a single line of code). Then it reiterates and the value is replaced.